redaxo/src/addons/media_manager/lib/effects/effect_workspace.php

```diff
@@ -46,45 +46,77 @@ public function execute()
             $trans = true;
         }
 
-        $workspace = imagecreatetruecolor($this->params['width'], $this->params['height']);
-        if ($trans) {
-            $transparent = imagecolorallocatealpha($workspace, 0, 0, 0, 127);
-            imagefill($workspace, 0, 0, $transparent);
-            $this->keepTransparent($workspace);
-        } else {
-            imagefill($workspace, 0, 0, imagecolorallocate($workspace, $this->params['bg_r'], $this->params['bg_g'], $this->params['bg_b']));
-        }
-
+		$workspace = imagecreatetruecolor($this->params['width'], $this->params['height']);
+		if ($trans) {
+			$transparent = imagecolorallocatealpha($workspace, 0, 0, 0, 127);
+			imagefill($workspace, 0, 0, $transparent);
+			$this->keepTransparent($workspace);
+		} else {
+			imagefill($workspace, 0, 0, imagecolorallocate($workspace, $this->params['bg_r'], $this->params['bg_g'], $this->params['bg_b']));
+		}
+		
         $srcW = $w;
         $srcH = $h;
         $dstX = 0;
         $dstY = 0;
         $srcX = 0;
         $srcY = 0;
-
+		$paddingX = 0;
+		$paddingY = 0;
+		$paramsHeight = (int) $this->params['height'];
+		$paramsWidth = (int) $this->params['width'];
+		
+		// Bild als Hintergrund ------------------------------
+
+		if ('image' == $this->params['set_transparent']) {
+			$bgimage = rex_path::media($this->params['bgimage']);
+			if (!is_file($bgimage)) {
+				return;
+			}
+			$bg = new rex_managed_media($bgimage);
+			$bg->asImage();
+			$workspace = $bg->getImage();
+			imagealphablending($workspace, true);
```

Comment:
Wäre hier nicht das besser?

```suggestion
			$this->keepTransparent($workspace);
```

Reply:
kannte ich noch nicht .... bewirkt wahrscheinlich dasselbe, oder?

Reply:
Ja, wobei dort noch mehr aufgerufen wird. Insbesondere gifs werden dort sonderbehandelt.
Wieso das ganze, weiß ich auch nicht mehr so im Detail. Aber die Methode rufen wir überall auf.

Reply:
... wieder watt gelernt ...